modules/doctor/storage.go

```diff
@@ -6,71 +6,247 @@ package doctor
 )
 
-func checkAttachmentStorageFiles(logger log.Logger, autofix bool) error {
-	var total, garbageNum int
-	var deletePaths []string
-	if err := storage.Attachments.IterateObjects(func(p string, obj storage.Object) error {
+type commonStorageCheckOptions struct {
+	storer       storage.ObjectStorage
+	isAssociated func(path string, obj storage.Object, stat fs.FileInfo) (bool, error)
+	name         string
+}
+
+func commonCheckStorage(ctx context.Context, logger log.Logger, autofix bool, opts *commonStorageCheckOptions) error {
+	totalCount, unassociatedCount := 0, 0
+	totalSize, unassociatedSize := int64(0), int64(0)
+
+	var pathsToDelete []string
+	if err := opts.storer.IterateObjects(func(p string, obj storage.Object) error {
 		defer obj.Close()
 
-		total++
+		totalCount++
 		stat, err := obj.Stat()
 		if err != nil {
 			return err
 		}
-		exist, err := repo_model.ExistAttachmentsByUUID(stat.Name())
+		totalSize += stat.Size()
+
+		associated, err := opts.isAssociated(p, obj, stat)
 		if err != nil {
 			return err
 		}
-		if !exist {
-			garbageNum++
+		if !associated {
+			unassociatedCount++
+			unassociatedSize += stat.Size()
 			if autofix {
-				deletePaths = append(deletePaths, p)
+				pathsToDelete = append(pathsToDelete, p)
 			}
 		}
 		return nil
 	}); err != nil {
-		logger.Error("storage.Attachments.IterateObjects failed: %v", err)
+		logger.Error("Error whilst iterating %s storage: %v", opts.name, err)
 		return err
 	}
 
-	if garbageNum > 0 {
+	if unassociatedCount > 0 {
 		if autofix {
 			var deletedNum int
-			for _, p := range deletePaths {
-				if err := storage.Attachments.Delete(p); err != nil {
-					log.Error("Delete attachment %s failed: %v", p, err)
+			for _, p := range pathsToDelete {
+				if err := opts.storer.Delete(p); err != nil {
+					log.Error("Error whilst deleting %s from %s storage: %v", p, opts.name, err)
 				} else {
 					deletedNum++
 				}
 			}
-			logger.Info("%d missed information attachment detected, %d deleted.", garbageNum, deletedNum)
+			logger.Info("Deleted %d/%d unassociated %s(s)", deletedNum, unassociatedCount, opts.name)
 		} else {
-			logger.Warn("Checked %d attachment, %d missed information.", total, garbageNum)
+			logger.Warn("Found %d/%d (%s/%s) unassociated %s(s)", unassociatedCount, totalCount, base.FileSize(unassociatedSize), base.FileSize(totalSize), opts.name)
 		}
+	} else {
+		logger.Info("Found %d (%s) %s(s)", totalCount, base.FileSize(totalSize), opts.name)
 	}
 	return nil
 }
 
-func checkStorageFiles(ctx context.Context, logger log.Logger, autofix bool) error {
-	if err := storage.Init(); err != nil {
-		logger.Error("storage.Init failed: %v", err)
-		return err
+type storageCheckOptions struct {
+	All          bool
+	Attachments  bool
+	LFS          bool
+	Avatars      bool
+	RepoAvatars  bool
+	RepoArchives bool
+	Packages     bool
+}
+
+func checkStorage(opts *storageCheckOptions) func(ctx context.Context, logger log.Logger, autofix bool) error {
```

Comment:
```suggestion
// checkStorage checks for all checked storages if they contain unadopted files
func checkStorage(opts *storageCheckOptions) func(ctx context.Context, logger log.Logger, autofix bool) error {
```

Reply:
I think we need to be very careful with the nomenclature here. I would very strongly argue that adding in `adoption` as an additional descriptive term is confusing.

I've used `associated` elsewhere in the code therefore this should use `associated`/`unassociated`.

We could change to use `orphaned` instead which would fit with the term used for inconsistent database entries.

but I would STRONGLY argue that we must NOT use `adopted` here. That term is used for something entirely diffferent.